kong/init.lua

```diff
@@ -742,6 +742,12 @@ function Kong.preread()
 
   runloop.preread.before(ctx)
 
+  -- if proxying to a second layer TLS terminator is required
+  -- abort further execution and return back to Nginx
+  if ctx.stream_proxy_preread_terminate then
```

Comment:
Instead of checking the flag, if we add a return value for `runloop.preread.before` does that makes more sense? Flags are slower but more important it makes the control flow less clear :)